examples/misc/generics/list.bi

```diff
@@ -0,0 +1,66 @@
+#macro list(T)
+	__fb_join__(list_,__fb_arg_rightof__(T,OF))
+#endmacro
+
+#macro list_of(T)
+	list_##T
+#endmacro
+
+#macro imp_list_of(T)
+#if not defined(list_##T)	
+	type list_item_##T
+		value as T
+		next_ as list_item_##T ptr
+	end type
+
+	type list_##T

```

Comment:
It would probably be good to also define the copy constructor and `operator let` (or at least declare them and leave them unimplemented).